ntoskrnl/ke/i386/mp.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Architecture source file to hold multiprocessor functions
+ * COPYRIGHT:   Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES *****************************************************************/
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS *******************************************************************/
+
+#define PHYSICAL_ADDRESS  LARGE_INTEGER
+PHYSICAL_ADDRESS HighestPhysicalAddress;
+ULONG ProcessorCount = 0;
+
+struct APInfo
+{
+    KPCR pcr;
+    KTSS tss;
+    ETHREAD thread;
+    KDESCRIPTOR GdtDesc;
+    KDESCRIPTOR IdtDesc;
```

Comment:
These are part of KPROCESSOR_STATE::SpecialRegisters

Reply:
See comment above.

Reply:
See my blog post from 6 years ago.

Reply:
an URL to it could help :P

Reply:
After having another look, I got aware that what has been tried to do here was to copy the GDT, not tthe KDESCRIPTOR. So the structure should rather contain a `KGDTENTRY Gdt[128];`

Reply:
Something like this, except that the actual number of entries should be deduced from the BSP' GDT.Limit value.

Reply:
According to a comment in freeldr, the GDT is always 128 entries. And that is what I've seen on Windows as well. No idea why, but "that's what Windows does".